src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversions.cs

```diff
@@ -57,13 +57,39 @@ public override Conversion GetMethodGroupDelegateConversion(BoundMethodGroup sou
 
             Debug.Assert(methodSymbol == ((NamedTypeSymbol)destination).DelegateInvokeMethod);
 
-            // If synthesizing a delegate with `params` array, check that `ParamArrayAttribute` is available.
-            if (methodSymbol.OriginalDefinition is SynthesizedDelegateInvokeMethod { Parameters: [.., { IsParams: true }] })
+            if (methodSymbol.OriginalDefinition is SynthesizedDelegateInvokeMethod invoke)
             {
-                Binder.GetWellKnownTypeMember(Compilation,
-                    WellKnownMember.System_ParamArrayAttribute__ctor,
-                    out var memberUseSiteInfo);
-                useSiteInfo.Add(memberUseSiteInfo);
+                // If synthesizing a delegate with `params` array, check that `ParamArrayAttribute` is available.
+                if (invoke.IsParams())
+                {
+                    Binder.AddUseSiteDiagnosticForSynthesizedAttribute(
+                        Compilation,
+                        WellKnownMember.System_ParamArrayAttribute__ctor,
+                        ref useSiteInfo);
+                }
+
+                // If synthesizing a delegate with `decimal`/`DateTime` default value,
+                // check that the corresponding `*ConstantAttribute` is available.
+                foreach (var p in invoke.Parameters)
+                {
+                    var defaultValue = p.ExplicitDefaultConstantValue;
+                    if (defaultValue != ConstantValue.NotAvailable)
+                    {
+                        WellKnownMember? member = defaultValue.SpecialType switch
+                        {
+                            SpecialType.System_Decimal => WellKnownMember.System_Runtime_CompilerServices_DecimalConstantAttribute__ctorByteByteInt32Int32Int32,
+                            SpecialType.System_DateTime => WellKnownMember.System_Runtime_CompilerServices_DateTimeConstantAttribute__ctor,
+                            _ => null
+                        };
+                        if (member != null)
+                        {
+                            Binder.AddUseSiteDiagnosticForSynthesizedAttribute(
+                                Compilation,
+                                member.Value,
```

Comment:
>Value [](http://example.com/codeflow?start=39&length=5)

GetValueOrDefault? This avoids another null check

Reply:
Wouldn't the default be `WellKnownMember.System_Math__RoundDouble` (the enum member with value 0)? I don't want to add use-site diagnostics for that.

Reply:
Oh, you didn't mean to get rid of the `if`, just call `GetValueOrDefault`, got it now.